entities/user.go

```diff
@@ -0,0 +1,17 @@
+package entities
+
+import (
+	"github.com/unicsmcr/hs_auth/utils"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+// User is the struct to store registered users
+type User struct {
+	ID            primitive.ObjectID `json:"_id" bson:"_id"`
+	Name          string             `json:"name" bson:"name" validate:"required"`
+	Email         string             `json:"email" bson:"email" validate:"required,email"`
+	Password      string             `json:"password" bson:"password" validate:"required,min=6,max=50"`
+	EmailVerified bool               `json:"email_verified,omitempty" bson:"email_verified,omitempty"`
+	AuthLevel     utils.AuthLevel    `json:"auth_level,omitempty" bson:"auth_level,omitempty" validate:"min=0"`
```

Comment:
Also, shouldn't `AuthLevel` be `validate:"required, min=0"`

Reply:
What `required` does is checks if the field has been initialized. In go all variables get set to their default values (the zero value) if they're not initialized. The zero value for an `int` is 0 which represents the base `AuthLevel`. So having `user.AuthLevel` optional makes the code a bit cleaner by allowing us to not declare the value of `user.AuthLevel` explicitly.

Also `required` would not allow us to use 0 for `AuthLevel` since it's the default value and the validation library assumes that the field just hasn't been initialized and throws an error.

That said, I should've probably defined an upper bound for the field as well.